tasks/eth/006-MCP-L1/SignFromJsonWithAssertions.s.sol

```diff
@@ -0,0 +1,296 @@
+import {OptimismPortal} from "@eth-optimism-bedrock/src/L1/OptimismPortal.sol";
+import {L1CrossDomainMessenger} from "@eth-optimism-bedrock/src/L1/L1CrossDomainMessenger.sol";
+import {OptimismMintableERC20Factory} from "@eth-optimism-bedrock/src/universal/OptimismMintableERC20Factory.sol";
+import {L1ERC721Bridge} from "@eth-optimism-bedrock/src/L1/L1ERC721Bridge.sol";
+import {Predeploys} from "@eth-optimism-bedrock/src/libraries/Predeploys.sol";
+import {Types} from "@eth-optimism-bedrock/scripts/Types.sol";
+import {Deployer} from "@eth-optimism-bedrock/scripts/Deployer.sol";
+import {console2 as console} from "forge-std/console2.sol";
+import {stdJson} from "forge-std/StdJson.sol";
+
+contract SignFromJsonWithAssertions is SignFromJson, Deployer {
+    Types.ContractSet proxies;
+
+    uint256 l2OutputOracleStartingTimestamp;
+
+    uint256 xdmSenderSlotNumber;
+
+    /// @notice Sets up the contract.
+    function setUp() public override {
+        super.setUp();
+
+        proxies = _getContractSet();
+
+        l2OutputOracleStartingTimestamp = cfg.l2OutputOracleStartingTimestamp();
+
+        // Read the slot number for the xDomainMsgSender in the L1CrossDomainMessenger
+        try vm.readFile(
+            string.concat(
+                vm.projectRoot(),
+                "/lib/optimism/packages/contracts-bedrock/snapshots/storageLayout/L1CrossDomainMessenger.json"
+            )
+        ) returns (string memory data) {
+            xdmSenderSlotNumber = stdJson.readUint(data, string(abi.encodePacked("$.[13].slot")));
+        } catch {
+            revert(
+                "Failed to read xDomainMsgSender slot number in lib/optimism/packages/contracts-bedrock/snapshots/storageLayout/L1CrossDomainMessenger.json"
+            );
+        }
+    }
+
+    /// @notice The name of the script
+    function name() public pure override returns (string memory) {
+        return "SignFromJsonWithAssertions";
+    }
+
+    /// @notice Asserts that the SystemConfig is setup correctly
+    function checkSystemConfig() internal view {
+        console.log("Running chain assertions on the SystemConfig");
+
+        require(proxies.SystemConfig.code.length != 0, "200");
+
+        SystemConfig config = SystemConfig(proxies.SystemConfig);
+
+        ResourceMetering.ResourceConfig memory resourceConfig = config.resourceConfig();
+
+        require(config.owner() == cfg.finalSystemOwner(), "300");
+        require(config.overhead() == cfg.gasPriceOracleOverhead(), "400");
+        require(config.scalar() == cfg.gasPriceOracleScalar(), "500");
+        require(config.batcherHash() == bytes32(uint256(uint160(cfg.batchSenderAddress()))), "600");
+        require(config.gasLimit() == uint64(cfg.l2GenesisBlockGasLimit()), "700");
+        require(config.unsafeBlockSigner() == cfg.p2pSequencerAddress(), "800");
+        // Check _config
+        ResourceMetering.ResourceConfig memory rconfig = Constants.DEFAULT_RESOURCE_CONFIG();
```

Comment:
We could hash the config here like the other check instead of checking individual properties, gives guarantee of exhaustiveness